<a href="https://colab.research.google.com/github/Jiraporn-klobrat/data_viz_2022/blob/main/Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Present Project กลุ่ม
ทำ Dashboard ของ Data Covid (ตั้งแต่วันที่ 12/01/2020 - 18/03/2022) เล่ามุมมองที่เราเลือกเอง + ใช้ข้อมูลภูมิภาค + ข้อมูลอื่นที่เลือกเอง

เริ่มพรีเซ็นต์ 25 มีค 30 มีค 1 เมษา

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import os 
path = '/content/drive/My Drive/Project_dataviz22'

In [ ]:
import pandas as pd
import numpy as np

คำอธิบาย
* No	รหัสผู้ป่วย
* Age	อายุของผู้ป่วย
* Sex	เพศผู้ป่วย
* Nationality	สัญชาติ
* Province of isolation	จังหวัดที่รักษาตัว
* Notification Date	วันที่กระทรวงได้รับรายงาน
* Announce Date	วันที่แถลงข่าว
* Province of onset		จังหวัดที่เริ่มพบว่าตัวเองมีอาการป่วย
* District of onset		อำเภอที่เริ่มพบว่าตัวเองมีอาการป่วย
* risk สถานที่


## ตารางข้อมูล covid ปี2020

In [ ]:
data_covid2020 =  pd.read_csv(os.path.join(path,'confirmed-cases.csv'))
data_covid2020

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


No. announce_date Notified date   sex   age Unit nationality  \
0            1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1            2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2            3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3            4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4            5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...        ...           ...           ...   ...   ...  ...         ...   
839766  839767     12/8/2021     11/8/2021   ชาย  52.0   ปี         NaN   
839767  839768     12/8/2021     11/8/2021   ชาย  34.0   ปี    Thailand   
839768  839769     12/8/2021     11/8/2021   ชาย  31.0   ปี    Thailand   
839769  839770     12/8/2021     11/8/2021   ชาย  83.0   ปี    Thailand   
839770  839771     12/8/2021     11/8/2021   ชาย  47.0   ปี    Thailand   

       province_of_isolation                              risk  \
0              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
1              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
2                     นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ   
3              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
4                    นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ   
...                      ...                               ...   
839766             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839767             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839768             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839769             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   
839770             กาญจนบุรี                  ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  
0          กรุงเทพมหานคร               NaN  
1          กรุงเทพมหานคร               NaN  
2                 นครปฐม             เมือง  
3          กรุงเทพมหานคร               NaN  
4          กรุงเทพมหานคร               NaN  
...                  ...               ...  
839766         กาญจนบุรี               NaN  
839767         กาญจนบุรี               NaN  
839768         กาญจนบุรี               NaN  
839769         กาญจนบุรี               NaN  
839770         กาญจนบุรี               NaN  

[839771 rows x 11 columns]

## ตารางข้อมูล covid ปี2021(1)

In [ ]:
data_covid2021_1 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-120864.csv'))
data_covid2021_1

No. announce_date Notified date   sex   age Unit nationality  \
0         816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1         816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2         816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3         816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4         816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
...          ...           ...           ...   ...   ...  ...         ...   
1042163  1859153    25/10/2021    24/10/2021   ชาย  45.0   ปี         NaN   
1042164  1859154    25/10/2021    24/10/2021   ชาย  31.0   ปี         NaN   
1042165  1859155    25/10/2021    24/10/2021   ชาย  31.0   ปี         NaN   
1042166  1859156    25/10/2021    24/10/2021   ชาย  24.0   ปี         NaN   
1042167  1859157    25/10/2021    24/10/2021   ชาย  25.0   ปี         NaN   

        province_of_isolation                                         risk  \
0                    เชียงราย                                        อื่นๆ   
1                    เชียงราย                                        อื่นๆ   
2                    เชียงราย                                        อื่นๆ   
3                    เชียงราย                                        อื่นๆ   
4                    เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                       ...                                          ...   
1042163         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042164         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042165         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042166         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
1042167             เชียงใหม่                             ทัณฑสถาน/เรือนจำ   

        province_of_onset district_of_onset  
0                     NaN               NaN  
1                     NaN               NaN  
2                     NaN               NaN  
3                     NaN               NaN  
4                     NaN               NaN  
...                   ...               ...  
1042163               NaN               NaN  
1042164               NaN               NaN  
1042165               NaN               NaN  
1042166               NaN               NaN  
1042167               NaN               NaN  

[1042168 rows x 11 columns]

## ตารางข้อมูล covid ปี2021(2)

In [ ]:
data_covid2021_2 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-271064.csv'))
data_covid2021_2

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


No. announce_date Notified date   sex   age Unit nationality  \
0        1859158    26/10/2021    25/10/2021   ชาย  20.0   ปี    Thailand   
1        1859159    26/10/2021    25/10/2021  หญิง  42.0   ปี    Thailand   
2        1859160    26/10/2021    25/10/2021  หญิง  33.0   ปี    Thailand   
3        1859161    26/10/2021    25/10/2021  หญิง  52.0   ปี    Thailand   
4        1859162    26/10/2021    25/10/2021  หญิง  84.0   ปี    Thailand   
...          ...           ...           ...   ...   ...  ...         ...   
1010454  2869612     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010455  2869613     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010456  2869614     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010457  2869615     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   
1010458  2869616     27/2/2022     26/2/2022   ชาย   NaN  NaN         NaN   

        province_of_isolation                                           risk  \
0                       ลำปาง                                          อื่นๆ   
1                       ลำปาง                                          อื่นๆ   
2                       ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
3                       ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
4                       ลำปาง    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                       ...                                            ...   
1010454         กรุงเทพมหานคร                               ทัณฑสถาน/เรือนจำ   
1010455         กรุงเทพมหานคร                               ทัณฑสถาน/เรือนจำ   
1010456               ปัตตานี                               ทัณฑสถาน/เรือนจำ   
1010457               ปัตตานี                               ทัณฑสถาน/เรือนจำ   
1010458               ปัตตานี                               ทัณฑสถาน/เรือนจำ   

        province_of_onset district_of_onset  Unnamed: 11  
0                   ลำปาง        เมืองลำปาง          NaN  
1                   ลำปาง        เมืองลำปาง          NaN  
2                   ลำปาง          ห้างฉัตร          NaN  
3                   ลำปาง          ห้างฉัตร          NaN  
4                   ลำปาง        เมืองลำปาง          NaN  
...                   ...               ...          ...  
1010454               NaN               NaN          NaN  
1010455               NaN               NaN          NaN  
1010456               NaN               NaN          NaN  
1010457               NaN               NaN          NaN  
1010458               NaN               NaN          NaN  

[1010459 rows x 12 columns]

## ตารางข้อมูล covid ปี2022

In [ ]:
data_covid2022 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-280265.csv'))
data_covid2022

No. announce_date Notified date   sex   age Unit nationality  \
0       2869617     28/2/2022     27/2/2022   ชาย  41.0   ปี    Thailand   
1       2869618     28/2/2022     27/2/2022  หญิง  45.0   ปี    Thailand   
2       2869619     28/2/2022     27/2/2022  หญิง  40.0   ปี    Thailand   
3       2869620     28/2/2022     27/2/2022   ชาย  44.0   ปี    Thailand   
4       2869621     28/2/2022     27/2/2022   ชาย  11.0   ปี    Thailand   
...         ...           ...           ...   ...   ...  ...         ...   
433548  3303165     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433549  3303166     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433550  3303167     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433551  3303168     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433552  3303169     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   

       province_of_isolation                                         risk  \
0                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
1                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
2                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
3                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
4                      พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
...                      ...                                          ...   
433548         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
433549         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
433550         กรุงเทพมหานคร                             ทัณฑสถาน/เรือนจำ   
433551                 สงขลา                             ทัณฑสถาน/เรือนจำ   
433552                 สงขลา                             ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  
0                  พะเยา        เมืองพะเยา  
1                  พะเยา        เมืองพะเยา  
2                  พะเยา        เมืองพะเยา  
3                  พะเยา        เมืองพะเยา  
4                  พะเยา        เมืองพะเยา  
...                  ...               ...  
433548               NaN               NaN  
433549               NaN               NaN  
433550               NaN               NaN  
433551               NaN               NaN  
433552               NaN               NaN  

[433553 rows x 11 columns]

### หาค่า min max ของ column NO. เพื่อดูว่าแต่ละข้อมูลมีต่าซำ้กันหรือไม่

In [ ]:
# หาค่า min max ของ column NO.
def min_max(i):
    a = min(i['No.'])
    b = max(i['No.'])
    print(f'min of NO. is {a}')
    print(f'max of NO. is {b}')

In [ ]:
min_max(data_covid2020)   

min of NO. is 1
max of NO. is 839771


In [ ]:
min_max(data_covid2021_1)   

min of NO. is 816990
max of NO. is 1859157


In [ ]:
min_max(data_covid2021_2)   

min of NO. is 1859158
max of NO. is 2869616


In [ ]:
min_max(data_covid2022)   

min of NO. is 2869617
max of NO. is 3303169


จะเห็นได้ว่าจากตาราง(data_covid2020) ตารางที่ 1 มีค่า max of NO. is 839771 ซึ่งค่าจากตาราง(data_covid2021_1) min of NO. is 816990 แสดงค่าทั้ง 2 ตารางมีค่าซำ้กัน





In [ ]:
data_covid2020.iloc[816989:,:].head(10)    # ดูข้อมูลของตารางที่ 1  ตั้งแต่แถวที่ 816989 ถึงแถวสุดท้าย

No. announce_date Notified date   sex   age Unit nationality  \
816989  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
816990  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
816991  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
816992  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
816993  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
816994  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
816995  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
816996  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
816997  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
816998  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

       province_of_isolation                                         risk  \
816989              เชียงราย                                        อื่นๆ   
816990              เชียงราย                                        อื่นๆ   
816991              เชียงราย                                        อื่นๆ   
816992              เชียงราย                                        อื่นๆ   
816993              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
816994              เชียงราย                                        อื่นๆ   
816995              เชียงราย                                        อื่นๆ   
816996              เชียงราย                                        อื่นๆ   
816997              เชียงราย                                        อื่นๆ   
816998              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

       province_of_onset district_of_onset  
816989               NaN               NaN  
816990               NaN               NaN  
816991               NaN               NaN  
816992               NaN               NaN  
816993               NaN               NaN  
816994               NaN               NaN  
816995               NaN               NaN  
816996               NaN               NaN  
816997               NaN               NaN  
816998               NaN               NaN

In [ ]:
data_covid2021_1.head(10)    # ดูข้อมูล covid_data2

No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
5  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
6  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
7  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
8  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
9  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
5              เชียงราย                                        อื่นๆ   
6              เชียงราย                                        อื่นๆ   
7              เชียงราย                                        อื่นๆ   
8              เชียงราย                                        อื่นๆ   
9              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  
5               NaN               NaN  
6               NaN               NaN  
7               NaN               NaN  
8               NaN               NaN  
9               NaN               NaN

จากตารางทั้งสองที่แสดงจะเห็นได้ว่ามีค่าซำ้กันจริง

ดังนั้นต้องทำการลบค่าที่ซำ้กันออก

In [ ]:
covid2020_dropped = data_covid2020.iloc[:816989,:]
covid2020_dropped 

No. announce_date Notified date   sex   age Unit nationality  \
0            1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1            2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2            3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3            4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4            5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...        ...           ...           ...   ...   ...  ...         ...   
816984  816985     11/8/2021     10/8/2021  หญิง   8.0   ปี    Thailand   
816985  816986     11/8/2021     10/8/2021   ชาย  14.0   ปี    Thailand   
816986  816987     11/8/2021     10/8/2021   ชาย  59.0   ปี    Thailand   
816987  816988     11/8/2021     10/8/2021  หญิง  34.0   ปี    Thailand   
816988  816989     11/8/2021     10/8/2021  หญิง  19.0   ปี    Thailand   

       province_of_isolation  \
0              กรุงเทพมหานคร   
1              กรุงเทพมหานคร   
2                     นครปฐม   
3              กรุงเทพมหานคร   
4                    นนทบุรี   
...                      ...   
816984               สระแก้ว   
816985               สระแก้ว   
816986               สระแก้ว   
816987               สระแก้ว   
816988               สระแก้ว   

                                                     risk province_of_onset  \
0                        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1                        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                        คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3                        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4                        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
...                                                   ...               ...   
816984                               อยู่ระหว่างการสอบสวน           สระแก้ว   
816985                               อยู่ระหว่างการสอบสวน           สระแก้ว   
816986  อาชีพเสี่ยง เช่น ทำงานในสถานที่แออัด หรือทำงาน...           สระแก้ว   
816987                                              อื่นๆ     กรุงเทพมหานคร   
816988                                              อื่นๆ   พระนครศรีอยุธยา   

       district_of_onset  
0                    NaN  
1                    NaN  
2                  เมือง  
3                    NaN  
4                    NaN  
...                  ...  
816984           ตาพระยา  
816985           ตาพระยา  
816986        อรัญประเทศ  
816987       บางขุนเทียน  
816988             อุทัย  

[816989 rows x 11 columns]

## ต่อตาราง


In [ ]:
covid = pd.concat([covid2020_dropped,data_covid2021_1,data_covid2021_2,data_covid2022])
covid

No. announce_date Notified date   sex   age Unit nationality  \
0             1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1             2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2             3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3             4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4             5     24/1/2020           NaN  หญิง  66.0   ปี       China   
...         ...           ...           ...   ...   ...  ...         ...   
433548  3303165     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433549  3303166     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433550  3303167     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433551  3303168     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   
433552  3303169     18/3/2022     17/3/2022  หญิง   NaN  NaN         NaN   

       province_of_isolation                              risk  \
0              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
1              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
2                     นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ   
3              กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ   
4                    นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ   
...                      ...                               ...   
433548         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
433549         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
433550         กรุงเทพมหานคร                  ทัณฑสถาน/เรือนจำ   
433551                 สงขลา                  ทัณฑสถาน/เรือนจำ   
433552                 สงขลา                  ทัณฑสถาน/เรือนจำ   

       province_of_onset district_of_onset  Unnamed: 11  
0          กรุงเทพมหานคร               NaN          NaN  
1          กรุงเทพมหานคร               NaN          NaN  
2                 นครปฐม             เมือง          NaN  
3          กรุงเทพมหานคร               NaN          NaN  
4          กรุงเทพมหานคร               NaN          NaN  
...                  ...               ...          ...  
433548               NaN               NaN          NaN  
433549               NaN               NaN          NaN  
433550               NaN               NaN          NaN  
433551               NaN               NaN          NaN  
433552               NaN               NaN          NaN  

[3303169 rows x 12 columns]

## เลือกเฉพาะคอลัมน์ที่สนใจ

In [ ]:
covid_data = covid[['announce_date','sex','age','nationality','risk','province_of_onset']]
covid_data

announce_date   sex   age nationality  \
0          12/1/2020  หญิง  61.0       China   
1          17/1/2020  หญิง  74.0       China   
2          22/1/2020  หญิง  73.0    Thailand   
3          22/1/2020   ชาย  68.0       China   
4          24/1/2020  หญิง  66.0       China   
...              ...   ...   ...         ...   
433548     18/3/2022  หญิง   NaN         NaN   
433549     18/3/2022  หญิง   NaN         NaN   
433550     18/3/2022  หญิง   NaN         NaN   
433551     18/3/2022  หญิง   NaN         NaN   
433552     18/3/2022  หญิง   NaN         NaN   

                                    risk province_of_onset  
0       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
1       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
2       คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม  
3       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
4       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
...                                  ...               ...  
433548                  ทัณฑสถาน/เรือนจำ               NaN  
433549                  ทัณฑสถาน/เรือนจำ               NaN  
433550                  ทัณฑสถาน/เรือนจำ               NaN  
433551                  ทัณฑสถาน/เรือนจำ               NaN  
433552                  ทัณฑสถาน/เรือนจำ               NaN  

[3303169 rows x 6 columns]

## ตรวจสอบค่า Missing

In [ ]:
covid_data.isnull().any() #ดูว่าแต่ละ column มีค่า missing มั้ย

announce_date        False
sex                   True
age                   True
nationality           True
risk                  True
province_of_onset     True
dtype: bool

In [ ]:
covid_data[['sex']] = covid_data[['sex']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ
covid_data[['nationality']] = covid_data[['nationality']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ
covid_data[['province_of_onset']] = covid_data[['province_of_onset']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ
covid_data[['risk']] = covid_data[['risk']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ

/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:3641: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]


In [ ]:
covid_data.isnull().any() #เช็คข้อมูลในแต่ละคอลัมน์อีกครั้ง

announce_date        False
sex                  False
age                   True
nationality          False
risk                 False
province_of_onset    False
dtype: bool

In [ ]:
set(covid_data['province_of_onset']) 

{'\tประจวบคีรีขันธ์',
 'USA',
 '\xa0กาญจนบุรี',
 '\xa0ชลบุรี',
 '\xa0ตราด',
 '\xa0นครปฐม',
 '\xa0สมุทรสงคราม',
 '\xa0สมุทรสาคร',
 '\xa0อุทัยธานี',
 'กทม',
 'กทม.',
 'กบินทร์บุรี',
 'กระบี่',
 'กระบี่ี่',
 'กรุงเทพ',
 'กรุงเทพมหานคร',
 'กรุงเทพมหานคร.',
 'กรุงเทพมหานคร. / พระนครศรีอยุธยา',
 'กรุงเทพมหานคร. / ศรีสะเกษ',
 'กรุงเทพฯ',
 'กะทู้',
 'กัมพูชา',
 'กาญจนบุรี',
 'กาญจบุรี',
 'กาญนบุรี',
 'กาฬสิน',
 'กาฬสินธิ์',
 'กาฬสินธุ์',
 'กาฬสินธุ์\xa0',
 'กาฬสินธ์',
 'กำเพงเพชร',
 'กำแพง',
 'กำแพงเพชร',
 'กำแพงเพชร\xa0',
 'กําแพงเพชร',
 'ขอนแก่น',
 'ขอนแก่น\xa0',
 'ขอนแก่่น',
 'ขัยนาท',
 'ขัยภูมิ',
 'ขานุวรลักษณ์บุรี',
 'คนไทยเดินทางกลับจากต่างประเทศ',
 'คลองเตย',
 'จ.กรุงเทพมหานคร',
 'จ.ภูเก็ต',
 'จ.ราชบุรี',
 'จ.สุราษฎร์ธานี',
 'จังหวัดสมุทรสาคร',
 'จันทบุรี',
 'ฉะเชองเทราร',
 'ฉะเชิงเทรา',
 'ชมพร',
 'ชลบุรี',
 'ชลบุรี\t',
 'ชลบุรี\xa0',
 'ชลบุุรี',
 'ชอนแก่น',
 'ชัยนาถ',
 'ชัยนาท',
 'ชัยบาดาล',
 'ชัยภูมิ',
 'ชัยภูมิ\xa0',
 'ชุมพร',
 'ชุุมพร',
 'ชุุุมพร',
 'ดอนมดแดง',
 'ดินแดง',
 'ตรัง',
 

In [ ]:
len(set(covid_data['province_of_onset'])) 

436

In [ ]:
#ทำข้อมูลให้เหมือนกัน
covid_data['province_of_onset'].replace({'\tประจวบคีรีขันธ์':'ประจวบคีรีขันธ์',
 'USA':'กรุงเทพมหานคร',
 '\xa0กาญจนบุรี':'กาญจนบุรี',
 '\xa0ชลบุรี':'ชลบุรี',
 '\xa0ตราด':'ตราด',
 '\xa0นครปฐม':'นครปฐม',
 '\xa0สมุทรสงคราม':'สมุทรสงคราม',
 '\xa0สมุทรสาคร':'สมุทรสาคร',
 '\xa0อุทัยธานี':'อุทัยธานี',
 'กทม':'กรุงเทพมหานคร',
 'กทม.':'กรุงเทพมหานคร',
 'กบินทร์บุรี':'ปราจีนบุรี',
 'กระบี่':'กระบี่',
 'กระบี่ี่':'กระบี่',
 'กรุงเทพ':'กรุงเทพมหานคร',
 'กรุงเทพมหานคร':'กรุงเทพมหานคร',
 'กรุงเทพมหานคร.':'กรุงเทพมหานคร',
 'กรุงเทพมหานคร. / พระนครศรีอยุธยา':'กรุงเทพมหานคร',
 'กรุงเทพมหานคร. / ศรีสะเกษ':'กรุงเทพมหานคร',
 'กรุงเทพฯ':'กรุงเทพมหานคร',
 'กะทู้':'ไม่ระบุ',
 'กัมพูชา':'กรุงเทพมหานคร',
 'กาญจนบุรี':'กาญจนบุรี',
 'กาญจบุรี':'กาญจนบุรี',
 'กาญนบุรี':'กาญจนบุรี',
 'กาฬสิน':'กาฬสินธุ์',
 'กาฬสินธิ์':'กาฬสินธุ์',
 'กาฬสินธุ์':'กาฬสินธุ์',
 'กาฬสินธุ์\xa0':'กาฬสินธุ์',
 'กาฬสินธ์':'กาฬสินธุ์',
 'กำเพงเพชร':'กำแพงเพชร',
 'กำแพง':'กำแพงเพชร',
 'กำแพงเพชร':'กำแพงเพชร',
 'กำแพงเพชร\xa0':'กำแพงเพชร',
 'กําแพงเพชร':'กำแพงเพชร',
 'ขอนแก่น':'ขอนแก่น',
 'ขอนแก่น\xa0':'ขอนแก่น',
 'ขอนแก่่น':'ขอนแก่น',
 'ขัยนาท':'ชัยนาท',
 'ขัยภูมิ':'ชัยภูมิ',
 'ขานุวรลักษณ์บุรี':'กำแพงเพชร',
 'คนไทยเดินทางกลับจากต่างประเทศ':'กรุงเทพมหานคร',
 'คลองเตย':'กรุงเทพมหานคร',
 'จ.กรุงเทพมหานคร':'กรุงเทพมหานคร',
 'จ.ภูเก็ต':'ภูเก็ต',
 'จ.ราชบุรี':'ราชบุรี',
 'จ.สุราษฎร์ธานี':'สุราษฎร์ธานี',
 'จังหวัดสมุทรสาคร':'สมุทรสาคร',
 'จันทบุรี':'จันทบุรี',
 'ฉะเชองเทราร':'ฉะเชิงเทรา',
 'ฉะเชิงเทรา':'ฉะเชิงเทรา',
 'ชมพร':'ชุมพร',
 'ชลบุรี':'ชลบุรี',
 'ชลบุรี\t':'ชลบุรี',
 'ชลบุรี\xa0':'ชลบุรี',
 'ชลบุุรี':'ชลบุรี',
 'ชอนแก่น':'ขอนแก่น',
 'ชัยนาถ':'ชัยนาท',
 'ชัยนาท':'ชัยนาท',
 'ชัยบาดาล':'ลพบุรี',
 'ชัยภูมิ':'ชัยภูมิ',
 'ชัยภูมิ\xa0':'ชัยภูมิ',
 'ชุมพร':'ชุมพร',
 'ชุุมพร':'ชุมพร',
 'ชุุุมพร':'ชุมพร',
 'ดอนมดแดง':'อุบลราชธานี',
 'ดินแดง':'กรุงเทพมหานคร',
 'ตรัง':'ตรัง',
 'ตรััง':'ตรัง',
 'ตราด':'ตราด',
 'ตราด\xa0':'ตราด',
 'ตาก':'ตาก',
 'ตากใบ':'นราธิวาส',
 'ต่างจังหวัด':'กรุงเทพมหานคร',
 'ทุ่งเสลี่ยม':'สุโขทัย',
 'ท้ายบ้าน':'สมุทรปราการ',
 'ธารโต':'ยะลา',
 'นคนปฐม':'นครปฐม',
 'นคปฐม':'นครปฐม',
 'นครนปฐม':'นครปฐม',
 'นครนาชสีมา':'นครราชสีมา',
 'นครนายก':'นครนายก',
 'นครปฐม':'นครปฐม',
 'นครปฐม\xa0':'นครปฐม',
 'นครปบม':'นครปฐม',
 'นครพนม':'นครพนม',
 'นครพนม\xa0':'นครพนม',
 'นครยายก':'นครนายก',
 'นครราขสีมา':'นครราชสีมา',
 'นครราชธานี':'นครราชสีมา',
 'นครราชศรีมา':'นครราชสีมา',
 'นครราชสสีมา':'นครราชสีมา',
 'นครราชสีมา':'นครราชสีมา',
 'นครราชสีมา ลพบุรี':'นครราชสีมา',
 'นครราชสีมา\xa0':'นครราชสีมา',
 'นครศรธรรมราช':'นครศรีธรรมราช',
 'นครศรี':'นครศรีธรรมราช',
 'นครศรีธรมมราช':'นครศรีธรรมราช',
 'นครศรีธรมราช':'นครศรีธรรมราช',
 'นครศรีธรรมราข':'นครศรีธรรมราช',
 'นครศรีธรรมราช':'นครศรีธรรมราช',
 'นครศรีฯ':'นครศรีธรรมราช',
 'นครศรีีธรรมราช':'นครศรีธรรมราช',
 'นครศร๊ธรรมราช':'นครศรีธรรมราช',
 'นครศร๊ะรรมราช':'นครศรีธรรมราช',
 'นครสรรค์':'นครสวรรค์',
 'นครสรีธรรมราช':'นครศรีธรรมราช',
 'นครสวรรค':'นครสวรรค์',
 'นครสวรรค์':'นครสวรรค์',
 'นครสวรรค์\xa0':'นครสวรรค์',
 'นครสีธรรมราช':'นครศรีธรรมราช',
 'นครฯ':'นครราชสีมา',
 'นคราชสีมา':'นครราชสีมา',
 'นครีศรีธรรมราช':'นครศรีธรรมราช',
 'นตรนายก':'นครนายก',
 'นตรปฐม':'นครปฐม',
 'นนทบุรี':'นนทบุรี',
 'นนทบุรี\xa0':'นนทบุรี',
 'นนทบุุรี':'นนทบุรี',
 'นนทุบรี':'นนทบุรี',
 'นนทุรี':'นนทบุรี',
 'นนมบุรี':'นนทบุรี',
 'นรศรีธรรมราช':'นครศรีธรรมราช',
 'นราธิวาส':'นราธิวาส',
 'นราธิวาส\u200b':'นราธิวาส',
 'นราธฺวาส':'นราธิวาส',
 'นอกเขต':'ไม่ระบุ',
 'นาราธิวาส':'นราธิวาส',
 'นาวัง':'หนองบัวลำภู',
 'น่าน':'น่าน',
 'น่าน/ฉะเชิงเทรา':'น่าน',
 'บางกรวย':'นนทบุรี',
 'บางกะปิ':'กรุงเทพมหานคร',
 'บางนา':'กรุงเทพมหานคร',
 'บางปะอิน':'กรุงเทพมหานคร',
 'บางละมุง':'ชลบุรี',
 'บางศรีเมือง':'นนทบุรี',
 'บางเพียง':'สมุทรปราการ',
 'บึงกาฬ':'บึงกาฬ',
 'บึงโขงหลง':'บึงกาฬ',
 'บุรีรัมภ์':'บุรีรัมย์',
 'บุรีรัมย์':'บุรีรัมย์',
 'บุรีรัมย์\xa0':'บุรีรัมย์',
 'บุรีรัยม์':'บุรีรัมย์',
 'บุรีรีมย์':'บุรีรัมย์',
 'บ้านหลวง':'น่าน',
 'ปจ':'ไม่ระบุ',
 'ปทึมธานี':'ปทุมธานี',
 'ปทุทธานี':'ปทุมธานี',
 'ปทุมฑานี':'ปทุมธานี',
 'ปทุมทานี':'ปทุมธานี',
 'ปทุมธาน':'ปทุมธานี',
 'ปทุมธานี':'ปทุมธานี',
 'ปทุมธานี.':'ปทุมธานี',
 'ปทุมธานี\xa0':'ปทุมธานี',
 'ปทุมธานีี':'ปทุมธานี',
 'ปทุมาธานี':'ปทุมธานี',
 'ปทุุมธานี':'ปทุมธานี',
 'ปธุมธานี':'ปทุมธานี',
 'ปมุมธานี':'ปทุมธานี',
 'ประจวนคีรีขันธ็':'ประจวบคีรีขันธ์',
 'ประจวบ':'ประจวบคีรีขันธ์',
 'ประจวบคิรีขันธ์':'ประจวบคีรีขันธ์',
 'ประจวบคีรขันธ์':'ประจวบคีรีขันธ์',
 'ประจวบคีรขัันธ์':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขัน':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขันธื':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขันธฺ':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขันธ์':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขันธ์\xa0':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขันธ์์':'ประจวบคีรีขันธ์',
 'ประจวบคีรีขัันธ์':'ประจวบคีรีขันธ์',
 'ประจวบคีีรีขันธ์':'ประจวบคีรีขันธ์',
 'ประจวบฯ':'ประจวบคีรีขันธ์',
 'ประทุมธานี':'ปทุมธานี',
 'ประเทศกัมพูชา':'กรุงเทพมหานคร',
 'ปราจีนบุรี':'ปราจีนบุรี',
 'ปราจีบุรี':'ปราจีนบุรี',
 'ปลวกแดง':'ระยอง',
 'ปลายพระยา':'กระบี่',
 'ปอยเปต':'กรุงเทพมหานคร',
 'ปัตตนรี':'ปัตตานี',
 'ปัตตานนี':'ปัตตานี',
 'ปัตตานี':'ปัตตานี',
 'ปัตตานีี':'ปัตตานี',
 'ปัตาตานี':'ปัตตานี',
 'ปัว':'เพชรบุรี',
 'ปััตตานี':'ปัตตานี',
 'ปุทมธานี':'ปทุมธานี',
 'ป่าโมก':'สุโขทัย',
 'พม่า':'กรุงเทพมหานคร',
 'พรหมพิราม':'พิษณุโลก',
 'พระนครศรียุธยา':'พระนครศรีอยุธยา',
 'พระนครศรีอยุทธยา':'พระนครศรีอยุธยา',
 'พระนครศรีอยุทยา':'พระนครศรีอยุธยา',
 'พระนครศรีอยุธยา':'พระนครศรีอยุธยา',
 'พระนครศรีอยุธยา\xa0':'พระนครศรีอยุธยา',
 'พะเยา':'พะเยา',
 'พังงา':'พังงา',
 'พัทยา':'ชลบุรี',
 'พัทลุง':'พัทลุง',
 'พัทลุุง':'พัทลุง',
 'พัังงา':'พังงา',
 'พิจิตร':'พิจิตร',
 'พิษณุโลก':'พิษณุโลก',
 'พิษณุโลก\xa0':'พิษณุโลก',
 'พิษณูโลก':'พิษณุโลก',
 'ฟินแลนด์':'กรุงเทพมหานคร',
 'ภาษีเจริญ':'กรุงเทพมหานคร',
 'ภุเก็ต':'ภูเก็ต',
 'ภูุเก็ต':'ภูเก็ต',
 'ภููเก็ต':'ภูเก็ต',
 'ภูเก็ต':'ภูเก็ต',
 'ภูเก็็ต':'ภูเก็ต',
 'ภูเพียง':'น่าน',
 'มหาสาคาม':'มหาสารคาม',
 'มหาสารคาม':'มหาสารคาม',
 'มาเลเซีย':'กรุงเทพมหานคร',
 'มุกดาหาร':'มุกดาหาร',
 'มุกาดหาร':'มุกดาหาร',
 'ยะลา':'ยะลา',
 'ยานนาวา':'กรุงเทพมหานคร',
 'ยโสธร':'ยโสธร',
 'ร.จ.จ.นราธิวาส':'นราธิวาส',
 'รอ':'ไม่ระบุ',
 'ระนอง':'ระนอง',
 'ระบุ':'ไม่ระบุ',
 'ระยอง':'ระยอง',
 'ระยอง\xa0':'ระยอง',
 'ระอยง':'ระยอง',
 'ราชบบุรี':'ราชบุรี',
 'ราชบุรี':'ราชบุรี',
 'ราชบุรีี':'ราชบุรี',
 'ราชบุรี\u200b':'ราชบุรี',
 'ราชุรี':'ราชบุรี',
 'ราชเทวี':'กรุงเทพมหานคร',
 'ร้องกวาง':'แพร่',
 'ร้อยเอ็ด':'ร้อยเอ็ด',
 'ลบุรี':'ลพบุรี',
 'ลพบุรี':'ลพบุรี',
 'ลพบุรี.':'ลพบุรี',
 'ลพบุรี\xa0':'ลพบุรี',
 'ลอง':'ไม่ระบุ',
 'ลาดกระบัง':'กรุงเทพมหานคร',
 'ลาดพร้าว':'กรุงเทพมหานคร',
 'ลาว':'กรุงเทพมหานคร',
 'ลำปาง':'ลำปาง',
 'ลำพูน':'ลำพูน',
 'ลำพูน\u200b':'ลำพูน',
 'ลำพููน':'ลำพูน',
 'ลำลูกกา':'กรุงเทพมหานคร',
 'วังชิ้น':'แพร่',
 'วังทอง':'พิษณุโลก',
 'วิเศษชัยชาญ':'อ่างทอง',
 'ศรีษะสะเกษ':'ศรีสะเกษ',
 'ศรีษะเกษ':'ศรีสะเกษ',
 'ศรีสระเกษ':'ศรีสะเกษ',
 'ศรีสะเกษ':'ศรีสะเกษ',
 'ศีรสะเกษ':'ศรีสะเกษ',
 'สกลนคร':'สกลนคร',
 'สครสวรรค์':'นครสวรรค์',
 'สงขลา':'สงขลา',
 'สงขลา\u200b':'สงขลา',
 'สตูล':'สตูล',
 'สถานที่ทำงาน':'ไม่ระบุ',
 'สพรรณบุรี':'สุพรรณบุรี',
 'สมถทรปราการ':'สมุทรปราการ',
 'สมทรสงคราม':'สมุทรสงคราม',
 'สมทรสาคร':'สมุทรสาคร',
 'สมุครปราการ':'สมุทรปราการ',
 'สมุครสาคร':'สมุทรสาคร',
 'สมุท':'สมุทรปราการ',
 'สมุทปราการ':'สมุทรปราการ',
 'สมุทรปรากร':'สมุทรปราการ',
 'สมุทรปรากรร':'สมุทรปราการ',
 'สมุทรปรากา':'สมุทรปราการ',
 'สมุทรปราการ':'สมุทรปราการ',
 'สมุทรปราการ\xa0':'สมุทรปราการ',
 'สมุทรศาคร':'สมุทรสาคร',
 'สมุทรสงคราม':'สมุทรสาคร',
 'สมุทรสาคร':'สมุทรสาคร',
 'สมุทรสาคร\xa0':'สมุทรสาคร',
 'สมุทรสาคา':'สมุทรสาคร',
 'สมุทรสาคาร':'สมุทรสาคร',
 'สมุทรสาทร':'สมุทรสาคร',
 'สมุทรสารคร':'สมุทรสาคร',
 'สมุทราปราการ':'สมุทรปราการ',
 'สมุทราสาคร':'สมุทรสาคร',
 'สมุทสงคราม':'สมุทรสงคราม',
 'สมุทสาคร':'สมุทรสาคร',
 'สมุย':'สุราษฎร์ธานี',
 'สมุุทรปราการ':'สมุทรปราการ',
 'สมุุทรสงคราม':'สมุทรสงคราม',
 'สมุุทรสาคร':'สมุทรสาคร',
 'สระก้ว':'สระแก้ว',
 'สระบุรี':'สระบุรี',
 'สระบุรี\xa0':'สระบุรี',
 'สระแก้ว':'สระแก้ว',
 'สรินทร์':'สุรินทร์',
 'สวิสเซอร์แลนด์':'กรุงเทพมหานคร',
 'สอง':'ไม่ระบุ',
 'สองแคว':'เชียงใหม่',
 'สะพานสูง':'กรุงเทพมหานคร',
 'สะแก้ว':'สระแก้ว',
 'สัตหีบ':'ชลบุรี',
 'สัมผัสผู้ป่วยยืนยัน COVID20':'ไม่ระบุ',
 'สามุทรสาคร':'สมุทรสาคร',
 'สารคาม':'มหาสารคาม',
 'สารภี':'เชียงใหม่',
 'สิงห์บุรี':'สิงห์บุรี',
 'สิงห์บุรี\xa0':'สิงห์บุรี',
 'สุทรปราการ':'สมุทรปราการ',
 'สุพรรญบุรี':'สุพรรณบุรี',
 'สุพรรณบุรั':'สุพรรณบุรี',
 'สุพรรณบุรี':'สุพรรณบุรี',
 'สุพรรณบุรี\xa0':'สุพรรณบุรี',
 'สุมทรปราการ':'สมุทรปราการ',
 'สุมทรสาคร':'สมุทรสาคร',
 'สุมุทรปราการ':'สมุทรปราการ',
 'สุมุทรสาคร':'สมุทรสาคร',
 'สุระบุรี':'สระบุรี',
 'สุราษ':'สุราษฎร์ธานี',
 'สุราษฎณ์ธานี':'สุราษฎร์ธานี',
 'สุราษฎธานี':'สุราษฎร์ธานี',
 'สุราษฎร์':'สุราษฎร์ธานี',
 'สุราษฎร์ธานี':'สุราษฎร์ธานี',
 'สุราษฎร์ฯ':'สุราษฎร์ธานี',
 'สุราษฎร์์ธานี':'สุราษฎร์ธานี',
 'สุราษฎ์ธานี':'สุราษฎร์ธานี',
 'สุราษฏร์ธานี':'สุราษฎร์ธานี',
 'สุราษธานี':'สุราษฎร์ธานี',
 'สุราษร์ฎธานี':'สุราษฎร์ธานี',
 'สุราษร์ธานี':'สุราษฎร์ธานี',
 'สุราษฯ':'สุราษฎร์ธานี',
 'สุรินทรื':'สุรินทร์',
 'สุรินทร์':'สุรินทร์',
 'สุรินทร์ 32150':'สุรินทร์',
 'สุรินทร์,รีสะเกษ':'สุรินทร์',
 'สุรินทร์\xa0':'สุรินทร์',
 'สุริมทร์':'สุรินทร์',
 'สุริะแก้ว':'สระแก้ว',
 'สุวรรณภูมิ':'กรุงเทพมหานคร',
 'สุุพรรณบุรี':'สุพรรณบุรี',
 'สุุราษฎร์ธานี':'สุราษฎร์ธานี',
 'สุุราษฏร์ธานี':'สุราษฎร์ธานี',
 'สุุราษฯ':'สุราษฎร์ธานี',
 'สุุรินทร์':'สุรินทร์',
 'สุุุโขทัย':'สุโขทัย',
 'สุโขทับ':'สุโขทัย',
 'สุโขทัย':'สุโขทัย',
 'สุโขทัย\xa0':'สุโขทัย',
 'สูงเม่น':'แพร่',
 'หนวงบัวลำภู':'หนองบัวลำภู',
 'หนองคาย':'หนองคาย',
 'หนองงคาย':'หนองคาย',
 'หนองบัวภู':'หนองบัวลำภู',
 'หนองบัวลำพูน':'หนองบัวลำภู',
 'หนองบัวลำภู':'หนองบัวลำภู',
 'หนองบัวลําภู':'หนองบัวลำภู',
 'หนองบััวลำภู':'หนองบัวลำภู',
 'หนองม่วงไข่':'แพร่',
 'หนอบัวลำภู':'หนองบัวลำภู',
 'หลายจังหวัด':'ไม่ระบุ',
 'หัวหิน':'ชลบุรี',
 'ห้วยขวาง':'กรุงเทพมหานคร',
 'อยุทธยา':'พระนครศรีอยุธยา',
 'อยุธยา':'พระนครศรีอยุธยา',
 'อยุุธยา':'พระนครศรีอยุธยา',
 'อำนาจเจริญ':'อำนาจเจริญ',
 'อำนาจเจิญ':'อำนาจเจริญ',
 'อำนาญเจริญ':'อำนาจเจริญ',
 'อำนาเจริญ':'อำนาจเจริญ',
 'อิสราเอล':'กรุงเทพมหานคร',
 'อุดธานี':'อุดรธานี',
 'อุดร':'อุดรธานี',
 'อุดรธาณี':'อุดรธานี',
 'อุดรธานี':'อุดรธานี',
 'อุดรธานี\xa0':'อุดรธานี',
 'อุดรธานีี':'อุดรธานี',
 'อุดรราชธานี':'อุดรธานี',
 'อุตรดิตถ์':'อุตรดิตถ์',
 'อุตรดิตถ์\xa0':'อุตรดิตถ์',
 'อุตรดิตถ์์':'อุตรดิตถ์',
 'อุตรดิษถ์':'อุตรดิตถ์',
 'อุตรดิิตถ์':'อุตรดิตถ์',
 'อุตรติดถ์':'อุตรดิตถ์',
 'อุทัยธานี':'อุทัยธานี',
 'อุทัยานี':'อุทัยธานี',
 'อุบลราขธานี':'อุบลราชธานี',
 'อุบลราชธานี':'อุบลราชธานี',
 'อุบลราชธานี\xa0':'อุบลราชธานี',
 'อุุครดิตถ์':'อุตรดิตถ์',
 'อุุดรธาน':'อุดรธานี',
 'อุุดรธานี':'อุดรธานี',
 'อุุตรดิตถ์':'อุตรดิตถ์',
 'อุุทัยธานี':'อุทัยธานี',
 'อุุบลราชธานี':'อุบลราชธานี',
 'อ่งทอง':'อ่างทอง',
 'อ่างทอง':'อ่างทอง',
 'อ่างทอง\xa0':'อ่างทอง',
 'อํานาจเจริญ':'อำนาจเจริญ',
 'เชิงเทรา':'ฉะเชิงเทรา',
 'เชียงราย':'เชียงราย',
 'เชียงใหม่':'เชียงใหม่',
 'เชียรใหญ่':'เชียงใหม่',
 'เชีียงราย':'เชียงราย',
 'เชีียงใหม่':'เชียงใหม่',
 'เด่นชัย':'แพร่',
 'เพขรบุรี':'เพชรบุรี',
 'เพขรบูรณื':'เพชรบูรณ์',
 'เพขรบูรณ์':'เพชรบูรณ์',
 'เพชรบุรณ์':'เพชรบูรณ์',
 'เพชรบุรี':'เพชรบุรี',
 'เพชรบุรึ':'เพชรบุรี',
 'เพชรบุุรี':'เพชรบุรี',
 'เพชรบุูรณ์':'เพชรบูรณ์',
 'เพชรบูรณ์':'เพชรบูรณ์',
 'เพชรบูรณ์\xa0':'เพชรบูรณ์',
 'เพชรบูร์':'เพชรบูรณ์',
 'เพชรบููรณ์':'เพชรบูรณ์',
 'เมือง':'ไม่ระบุ',
 'เมืองตาก':'ตาก',
 'เมืองน่าน':'น่าน',
 'เมืองปัตตานี':'ปัตตานี',
 'เรือ':'ไม่ระบุ',
 'เลย':'เลย',
 'เลย\xa0':'เลย',
 'เวียงจันทร์':'กรุงเทพมหานคร',
 'เวียงสา':'กรุงเทพมหานคร',
 'แพร่':'แพร่',
 'แม่ทะ':'ลำปาง',
 'แม่ฮ่องสอน':'แม่ฮ่องสอน',
 'แล้วป่วย':'ไม่ระบุ',
 'โพธิ์ทอง':'อ่างทอง',
 'ไชโย':'อ่างทอง',
 'ไม่มีชื่อหอพัก':'ไม่ระบุ',
 'ไม่ระบุ':'ไม่ระบุ'}, inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/generic.py:6619: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return self._update_inplace(result)


In [ ]:
len(set(covid_data['province_of_onset'])) #มีค่าที่ชื่อว่า ไม่ระบุ

78

In [ ]:
data_gb = covid_data.groupby('province_of_onset').size().reset_index(name='จำนวนผู้ติดเชื้อ')  ### สร้างตารางที่รวมจำนวนผู้ป่วยในแต่ละจังหวัด
data_gb

province_of_onset  จำนวนผู้ติดเชื้อ
0             กระบี่             17641
1      กรุงเทพมหานคร            567335
2          กาญจนบุรี             24322
3          กาฬสินธุ์             18589
4          กำแพงเพชร             13042
..               ...               ...
73         เพชรบูรณ์             18362
74               เลย              9926
75              แพร่              5676
76        แม่ฮ่องสอน              4145
77           ไม่ระบุ            398327

[78 rows x 2 columns]

In [ ]:
covid_dropped = covid_data.dropna() #drop ค่าอายุที่ว่างออกจากตาราง

In [ ]:
covid_dropped 

announce_date      sex   age nationality  \
0          12/1/2020     หญิง  61.0       China   
1          17/1/2020     หญิง  74.0       China   
2          22/1/2020     หญิง  73.0    Thailand   
3          22/1/2020      ชาย  68.0       China   
4          24/1/2020     หญิง  66.0       China   
...              ...      ...   ...         ...   
433445     18/3/2022  ไม่ระบุ  43.0     ไม่ระบุ   
433446     18/3/2022  ไม่ระบุ  21.0     ไม่ระบุ   
433447     18/3/2022  ไม่ระบุ  22.0     ไม่ระบุ   
433448     18/3/2022  ไม่ระบุ  25.0     ไม่ระบุ   
433449     18/3/2022  ไม่ระบุ  64.0     ไม่ระบุ   

                                    risk province_of_onset  
0       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
1       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
2       คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม  
3       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
4       คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร  
...                                  ...               ...  
433445                             อื่นๆ           ไม่ระบุ  
433446                             อื่นๆ     กรุงเทพมหานคร  
433447                             อื่นๆ     กรุงเทพมหานคร  
433448                             อื่นๆ     กรุงเทพมหานคร  
433449                             อื่นๆ         กำแพงเพชร  

[3036374 rows x 6 columns]

In [ ]:
covid_finish = covid_dropped.merge(data_gb,how='left',left_on='province_of_onset',right_on = 'province_of_onset') 
covid_finish

announce_date      sex   age nationality  \
0           12/1/2020     หญิง  61.0       China   
1           17/1/2020     หญิง  74.0       China   
2           22/1/2020     หญิง  73.0    Thailand   
3           22/1/2020      ชาย  68.0       China   
4           24/1/2020     หญิง  66.0       China   
...               ...      ...   ...         ...   
3036369     18/3/2022  ไม่ระบุ  43.0     ไม่ระบุ   
3036370     18/3/2022  ไม่ระบุ  21.0     ไม่ระบุ   
3036371     18/3/2022  ไม่ระบุ  22.0     ไม่ระบุ   
3036372     18/3/2022  ไม่ระบุ  25.0     ไม่ระบุ   
3036373     18/3/2022  ไม่ระบุ  64.0     ไม่ระบุ   

                                     risk province_of_onset  จำนวนผู้ติดเชื้อ  
0        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335  
1        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335  
2        คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม             69534  
3        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335  
4        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335  
...                                   ...               ...               ...  
3036369                             อื่นๆ           ไม่ระบุ            398327  
3036370                             อื่นๆ     กรุงเทพมหานคร            567335  
3036371                             อื่นๆ     กรุงเทพมหานคร            567335  
3036372                             อื่นๆ     กรุงเทพมหานคร            567335  
3036373                             อื่นๆ         กำแพงเพชร             13042  

[3036374 rows x 7 columns]

# สร้างตารางภูมิภาค

In [ ]:
data_region = {'Province':['อำนาจเจริญ','บึงกาฬ','บุรีรัมย์','ชัยภูมิ','กาฬสินธุ์','ขอนแก่น','เลย','มหาสารคาม','มุกดาหาร','นครพนม','นครราชสีมา','หนองบัวลำภู','หนองคาย','ร้อยเอ็ด','สกลนคร','ศรีสะเกษ','สุรินทร์','อุบลราชธานี','อุดรธานี','ยโสธร','เชียงใหม่','เชียงราย','ลำปาง','ลำพูน','แม่ฮ่องสอน','น่าน','พะเยา','แพร่','อุตรดิตถ์','ตาก','สุโขทัย','พิษณุโลก','พิจิตร','กำแพงเพชร','เพชรบูรณ์','นครสวรรค์','อุทัยธานี','อ่างทอง','ชัยนาท','พระนครศรีอยุธยา','กรุงเทพมหานคร','ลพบุรี','นครปฐม','นนทบุรี','ปทุมธานี','สมุทรปราการ','สมุทรสาคร','สมุทรสงคราม','สระบุรี','สิงห์บุรี','สุพรรณบุรี','นครนายก','ฉะเชิงเทรา','จันทบุรี','ชลบุรี','ปราจีนบุรี','ระยอง','สระแก้ว','ตราด','กาญจนบุรี','ราชบุรี','เพชรบุรี','ประจวบคีรีขันธ์','ชุมพร','นครศรีธรรมราช','นราธิวาส','ปัตตานี','พัทลุง','สงขลา','สุราษฎร์ธานี','ยะลา','กระบี่','พังงา','ภูเก็ต','ระนอง','สตูล','ตรัง'],
             '6 ภูมิภาค (ภูมิศาสตร์)':['ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','ตะวันออกเฉียงเหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','เหนือ','ตะวันตก','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','กลาง','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันออก','ตะวันตก','ตะวันตก','ตะวันตก','ตะวันตก','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้','ใต้'],
}

In [ ]:
data_region = pd.DataFrame.from_dict(data_region)
data_region

Province 6 ภูมิภาค (ภูมิศาสตร์)
0   อำนาจเจริญ     ตะวันออกเฉียงเหนือ
1       บึงกาฬ     ตะวันออกเฉียงเหนือ
2    บุรีรัมย์     ตะวันออกเฉียงเหนือ
3      ชัยภูมิ     ตะวันออกเฉียงเหนือ
4    กาฬสินธุ์     ตะวันออกเฉียงเหนือ
..         ...                    ...
72       พังงา                    ใต้
73      ภูเก็ต                    ใต้
74       ระนอง                    ใต้
75        สตูล                    ใต้
76        ตรัง                    ใต้

[77 rows x 2 columns]

## merge ตาราง covid กับตาราง ภูมิภาค
covid_dropped กับ data_region

In [ ]:
data_cr = covid_finish.merge(data_region,how='left',left_on='province_of_onset',right_on = 'Province') 
data_cr

announce_date      sex   age nationality  \
0           12/1/2020     หญิง  61.0       China   
1           17/1/2020     หญิง  74.0       China   
2           22/1/2020     หญิง  73.0    Thailand   
3           22/1/2020      ชาย  68.0       China   
4           24/1/2020     หญิง  66.0       China   
...               ...      ...   ...         ...   
3036369     18/3/2022  ไม่ระบุ  43.0     ไม่ระบุ   
3036370     18/3/2022  ไม่ระบุ  21.0     ไม่ระบุ   
3036371     18/3/2022  ไม่ระบุ  22.0     ไม่ระบุ   
3036372     18/3/2022  ไม่ระบุ  25.0     ไม่ระบุ   
3036373     18/3/2022  ไม่ระบุ  64.0     ไม่ระบุ   

                                     risk province_of_onset  จำนวนผู้ติดเชื้อ  \
0        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
1        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
2        คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม             69534   
3        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
4        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
...                                   ...               ...               ...   
3036369                             อื่นๆ           ไม่ระบุ            398327   
3036370                             อื่นๆ     กรุงเทพมหานคร            567335   
3036371                             อื่นๆ     กรุงเทพมหานคร            567335   
3036372                             อื่นๆ     กรุงเทพมหานคร            567335   
3036373                             อื่นๆ         กำแพงเพชร             13042   

              Province 6 ภูมิภาค (ภูมิศาสตร์)  
0        กรุงเทพมหานคร                   กลาง  
1        กรุงเทพมหานคร                   กลาง  
2               นครปฐม                   กลาง  
3        กรุงเทพมหานคร                   กลาง  
4        กรุงเทพมหานคร                   กลาง  
...                ...                    ...  
3036369            NaN                    NaN  
3036370  กรุงเทพมหานคร                   กลาง  
3036371  กรุงเทพมหานคร                   กลาง  
3036372  กรุงเทพมหานคร                   กลาง  
3036373      กำแพงเพชร                   กลาง  

[3036374 rows x 9 columns]

In [ ]:
data_cr[['Province']] = data_cr[['Province']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ
data_cr[['6 ภูมิภาค (ภูมิศาสตร์)']] = data_cr[['6 ภูมิภาค (ภูมิศาสตร์)']].fillna('ไม่ระบุ') ##แทนค่า NaN เป็น ไม่ระบุ

In [ ]:
data_cr

announce_date      sex   age nationality  \
0           12/1/2020     หญิง  61.0       China   
1           17/1/2020     หญิง  74.0       China   
2           22/1/2020     หญิง  73.0    Thailand   
3           22/1/2020      ชาย  68.0       China   
4           24/1/2020     หญิง  66.0       China   
...               ...      ...   ...         ...   
3036369     18/3/2022  ไม่ระบุ  43.0     ไม่ระบุ   
3036370     18/3/2022  ไม่ระบุ  21.0     ไม่ระบุ   
3036371     18/3/2022  ไม่ระบุ  22.0     ไม่ระบุ   
3036372     18/3/2022  ไม่ระบุ  25.0     ไม่ระบุ   
3036373     18/3/2022  ไม่ระบุ  64.0     ไม่ระบุ   

                                     risk province_of_onset  จำนวนผู้ติดเชื้อ  \
0        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
1        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
2        คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม             69534   
3        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
4        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
...                                   ...               ...               ...   
3036369                             อื่นๆ           ไม่ระบุ            398327   
3036370                             อื่นๆ     กรุงเทพมหานคร            567335   
3036371                             อื่นๆ     กรุงเทพมหานคร            567335   
3036372                             อื่นๆ     กรุงเทพมหานคร            567335   
3036373                             อื่นๆ         กำแพงเพชร             13042   

              Province 6 ภูมิภาค (ภูมิศาสตร์)  
0        กรุงเทพมหานคร                   กลาง  
1        กรุงเทพมหานคร                   กลาง  
2               นครปฐม                   กลาง  
3        กรุงเทพมหานคร                   กลาง  
4        กรุงเทพมหานคร                   กลาง  
...                ...                    ...  
3036369        ไม่ระบุ                ไม่ระบุ  
3036370  กรุงเทพมหานคร                   กลาง  
3036371  กรุงเทพมหานคร                   กลาง  
3036372  กรุงเทพมหานคร                   กลาง  
3036373      กำแพงเพชร                   กลาง  

[3036374 rows x 9 columns]

## นำเข้าข้อมูลที่เลือกเอง

ข้อมูลสถิติจำนวนประชากรตามจังหวัด ปี 2564

In [ ]:
data_person = pd.read_excel(os.path.join(path,'stat_parson64.xlsx'))    # ไฟล์ที่เราต้องการ  # .read_excel อ่านไฟล์ที่เป็น excel  
data_person

ปีเดือน รหัสจังหวัด         ชื่อจังหวัด รหัสสำนักทะเบียน ชื่อสำนักทะเบียน  \
0     6412           0          ทั่วประเทศ                0                -   
1      6412          10       กรุงเทพมหานคร                0                -   
2      6412          11  จังหวัดสมุทรปราการ                0                -   
3      6412          12      จังหวัดนนทบุรี                0                -   
4      6412          13     จังหวัดปทุมธานี                0                -   
..      ...         ...                 ...              ...              ...   
231    6412          92         จังหวัดตรัง                0                -   
232    6412          93       จังหวัดพัทลุง                0                -   
233    6412          94      จังหวัดปัตตานี                0                -   
234    6412          95         จังหวัดยะลา                0                -   
235    6412          96     จังหวัดนราธิวาส                0                -   

    รหัสตำบล ชื่อตำบล รหัสหมู่บ้าน ชื่อหมู่บ้าน จำนวนประชากรชาย  \
0          0        -            0            -        32339118   
1          0        -            0            -         2592292   
2          0        -            0            -          645884   
3          0        -            0            -          599167   
4          0        -            0            -          563851   
..       ...      ...          ...          ...             ...   
231        0        -            0            -          312693   
232        0        -            0            -          254425   
233        0        -            0            -          360348   
234        0        -            0            -          269907   
235        0        -            0            -          400501   

    จำนวนประชากรหญิง จำนวนประชากรทั้งหมด จำนวนบ้าน (หลังคาเรือน)  
0           33832321            66171439                27708635  
1            2935702             5527994                 3147231  
2             710565             1356449                  733185  
3             689470             1288637                  728953  
4             626209             1190060                  660020  
..               ...                 ...                     ...  
231           327095              639788                  237935  
232           268116              522541                  199687  
233           369233              729581                  199103  
234           272407              542314                  173226  
235           409159              809660                  223141  

[236 rows x 13 columns]

In [ ]:
data_person_1= data_person[['ชื่อจังหวัด','จำนวนประชากรชาย','จำนวนประชากรหญิง']]
data_person_1

ชื่อจังหวัด จำนวนประชากรชาย จำนวนประชากรหญิง
0            ทั่วประเทศ        32339118         33832321
1         กรุงเทพมหานคร         2592292          2935702
2    จังหวัดสมุทรปราการ          645884           710565
3        จังหวัดนนทบุรี          599167           689470
4       จังหวัดปทุมธานี          563851           626209
..                  ...             ...              ...
231         จังหวัดตรัง          312693           327095
232       จังหวัดพัทลุง          254425           268116
233      จังหวัดปัตตานี          360348           369233
234         จังหวัดยะลา          269907           272407
235     จังหวัดนราธิวาส          400501           409159

[236 rows x 3 columns]

In [ ]:
data_person_1.isnull().any() #ดูว่าแต่ละ column มีค่า missing มั้ย

ชื่อจังหวัด         False
จำนวนประชากรชาย     False
จำนวนประชากรหญิง    False
dtype: bool

In [ ]:
set(data_person_1['ชื่อจังหวัด']) 

{'กรุงเทพมหานคร',
 'จังหวัดกระบี่',
 'จังหวัดกาญจนบุรี',
 'จังหวัดกาฬสินธุ์',
 'จังหวัดกำแพงเพชร',
 'จังหวัดขอนแก่น',
 'จังหวัดจันทบุรี',
 'จังหวัดฉะเชิงเทรา',
 'จังหวัดชลบุรี',
 'จังหวัดชัยนาท',
 'จังหวัดชัยภูมิ',
 'จังหวัดชุมพร',
 'จังหวัดตรัง',
 'จังหวัดตราด',
 'จังหวัดตาก',
 'จังหวัดนครนายก',
 'จังหวัดนครปฐม',
 'จังหวัดนครพนม',
 'จังหวัดนครราชสีมา',
 'จังหวัดนครศรีธรรมราช',
 'จังหวัดนครสวรรค์',
 'จังหวัดนนทบุรี',
 'จังหวัดนราธิวาส',
 'จังหวัดน่าน',
 'จังหวัดบึงกาฬ',
 'จังหวัดบุรีรัมย์',
 'จังหวัดปทุมธานี',
 'จังหวัดประจวบคีรีขันธ์',
 'จังหวัดปราจีนบุรี',
 'จังหวัดปัตตานี',
 'จังหวัดพระนครศรีอยุธยา',
 'จังหวัดพะเยา',
 'จังหวัดพังงา',
 'จังหวัดพัทลุง',
 'จังหวัดพิจิตร',
 'จังหวัดพิษณุโลก',
 'จังหวัดภูเก็ต',
 'จังหวัดมหาสารคาม',
 'จังหวัดมุกดาหาร',
 'จังหวัดยะลา',
 'จังหวัดยโสธร',
 'จังหวัดระนอง',
 'จังหวัดระยอง',
 'จังหวัดราชบุรี',
 'จังหวัดร้อยเอ็ด',
 'จังหวัดลพบุรี',
 'จังหวัดลำปาง',
 'จังหวัดลำพูน',
 'จังหวัดศรีสะเกษ',
 'จังหวัดสกลนคร',
 'จังหวัดสงขลา',
 'จังหวัดสตูล',
 'จังหวัดสม

In [ ]:
#ทำข้อมูลให้เหมือนกัน
data_person_1['ชื่อจังหวัด'].replace({'จังหวัดกระบี่':'กระบี่',
 'จังหวัดกาญจนบุรี':'กาญจนบุรี',
 'จังหวัดกาฬสินธุ์':'กาฬสินธุ์',
 'จังหวัดกำแพงเพชร':'กำแพงเพชร',
 'จังหวัดขอนแก่น':'ขอนแก่น',
 'จังหวัดจันทบุรี':'จันทบุรี',
 'จังหวัดฉะเชิงเทรา':'ฉะเชิงเทรา',
 'จังหวัดชลบุรี':'ชลบุรี',
 'จังหวัดชัยนาท':'ชัยนาท',
 'จังหวัดชัยภูมิ':'ชัยภูมิ',
 'จังหวัดชุมพร':'ชุมพร',
 'จังหวัดตรัง':'ตรัง',
 'จังหวัดตราด':'ตราด',
 'จังหวัดตาก':'ตาก',
 'จังหวัดนครนายก':'นครนายก',
 'จังหวัดนครปฐม':'นครปฐม',
 'จังหวัดนครพนม':'นครพนม',
 'จังหวัดนครราชสีมา':'นครราชสีมา',
 'จังหวัดนครศรีธรรมราช':'นครศรีธรรมราช',
 'จังหวัดนครสวรรค์':'นครสวรรค์',
 'จังหวัดนนทบุรี':'นนทบุรี',
 'จังหวัดนราธิวาส':'นราธิวาส',
 'จังหวัดน่าน':'น่าน',
 'จังหวัดบึงกาฬ':'บึงกาฬ',
 'จังหวัดบุรีรัมย์':'บุรีรัมย์',
 'จังหวัดปทุมธานี':'ปทุมธานี',
 'จังหวัดประจวบคีรีขันธ์':'ประจวบคีรีขันธ์',
 'จังหวัดปราจีนบุรี':'ปราจีนบุรี',
 'จังหวัดปัตตานี':'ปัตตานี',
 'จังหวัดพระนครศรีอยุธยา':'พระนครศรีอยุธยา',
 'จังหวัดพะเยา':'พะเยา',
 'จังหวัดพังงา':'พังงา',
 'จังหวัดพัทลุง':'พัทลุง',
 'จังหวัดพิจิตร':'พิจิตร',
 'จังหวัดพิษณุโลก':'พิษณุโลก',
 'จังหวัดภูเก็ต':'ภูเก็ต',
 'จังหวัดมหาสารคาม':'มหาสารคาม',
 'จังหวัดมุกดาหาร':'มุกดาหาร',
 'จังหวัดยะลา':'ยะลา',
 'จังหวัดยโสธร':'ยโสธร',
 'จังหวัดระนอง':'ระนอง',
 'จังหวัดระยอง':'ระยอง',
 'จังหวัดราชบุรี':'ราชบุรี',
 'จังหวัดร้อยเอ็ด':'ร้อยเอ็ด',
 'จังหวัดลพบุรี':'ลพบุรี',
 'จังหวัดลำปาง':'ลำปาง',
 'จังหวัดลำพูน':'ลำพูน',
 'จังหวัดศรีสะเกษ':'ศรีสะเกษ',
 'จังหวัดสกลนคร':'สกลนคร',
 'จังหวัดสงขลา':'สงขลา',
 'จังหวัดสตูล':'สตูล',
 'จังหวัดสมุทรปราการ':'สมุทรปราการ',
 'จังหวัดสมุทรสงคราม':'สมุทรสงคราม',
 'จังหวัดสมุทรสาคร':'สมุทรสาคร',
 'จังหวัดสระบุรี':'สระบุรี',
 'จังหวัดสระแก้ว':'สระแก้ว',
 'จังหวัดสิงห์บุรี':'สิงห์บุรี',
 'จังหวัดสุพรรณบุรี':'สุพรรณบุรี',
 'จังหวัดสุราษฎร์ธานี':'สุราษฎร์ธานี',
 'จังหวัดสุรินทร์':'สุรินทร์',
 'จังหวัดสุโขทัย':'สุโขทัย',
 'จังหวัดหนองคาย':'หนองคาย',
 'จังหวัดหนองบัวลำภู':'หนองบัวลำภู',
 'จังหวัดอำนาจเจริญ':'อำนาจเจริญ',
 'จังหวัดอุดรธานี':'อุดรธานี',
 'จังหวัดอุตรดิตถ์':'อุตรดิตถ์',
 'จังหวัดอุทัยธานี':'อุทัยธานี',
 'จังหวัดอุบลราชธานี':'อุบลราชธานี',
 'จังหวัดอ่างทอง':'อ่างทอง',
 'จังหวัดเชียงราย':'เชียงราย',
 'จังหวัดเชียงใหม่':'เชียงใหม่',
 'จังหวัดเพชรบุรี':'เพชรบุรี',
 'จังหวัดเพชรบูรณ์':'เพชรบูรณ์',
 'จังหวัดเลย':'เลย',
 'จังหวัดแพร่':'แพร่',
 'จังหวัดแม่ฮ่องสอน':'แม่ฮ่องสอน'}, inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/generic.py:6619: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return self._update_inplace(result)


In [ ]:
data_person_1

ชื่อจังหวัด จำนวนประชากรชาย จำนวนประชากรหญิง
0       ทั่วประเทศ        32339118         33832321
1    กรุงเทพมหานคร         2592292          2935702
2      สมุทรปราการ          645884           710565
3          นนทบุรี          599167           689470
4         ปทุมธานี          563851           626209
..             ...             ...              ...
231           ตรัง          312693           327095
232         พัทลุง          254425           268116
233        ปัตตานี          360348           369233
234           ยะลา          269907           272407
235       นราธิวาส          400501           409159

[236 rows x 3 columns]

## รวมตาราง data_cr (ข้อมูลโควิดกับภูมิภาค) กับ data_person_1

In [ ]:
data_cr_new = data_cr[['announce_date','sex','age','risk','province_of_onset','จำนวนผู้ติดเชื้อ','6 ภูมิภาค (ภูมิศาสตร์)']]
data_cr_new

announce_date      sex   age                              risk  \
0           12/1/2020     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1           17/1/2020     หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2           22/1/2020     หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3           22/1/2020      ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4           24/1/2020     หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
...               ...      ...   ...                               ...   
3036369     18/3/2022  ไม่ระบุ  43.0                             อื่นๆ   
3036370     18/3/2022  ไม่ระบุ  21.0                             อื่นๆ   
3036371     18/3/2022  ไม่ระบุ  22.0                             อื่นๆ   
3036372     18/3/2022  ไม่ระบุ  25.0                             อื่นๆ   
3036373     18/3/2022  ไม่ระบุ  64.0                             อื่นๆ   

        province_of_onset  จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์)  
0           กรุงเทพมหานคร            567335                   กลาง  
1           กรุงเทพมหานคร            567335                   กลาง  
2                  นครปฐม             69534                   กลาง  
3           กรุงเทพมหานคร            567335                   กลาง  
4           กรุงเทพมหานคร            567335                   กลาง  
...                   ...               ...                    ...  
3036369           ไม่ระบุ            398327                ไม่ระบุ  
3036370     กรุงเทพมหานคร            567335                   กลาง  
3036371     กรุงเทพมหานคร            567335                   กลาง  
3036372     กรุงเทพมหานคร            567335                   กลาง  
3036373         กำแพงเพชร             13042                   กลาง  

[3036374 rows x 7 columns]

In [ ]:
data_all = data_cr_new.merge(data_person_1,how='left',left_on='province_of_onset',right_on = 'ชื่อจังหวัด') 
data_all

announce_date      sex   age                              risk  \
0           12/1/2020     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1           12/1/2020     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2           12/1/2020     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3           17/1/2020     หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4           17/1/2020     หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
...               ...      ...   ...                               ...   
8558209     18/3/2022  ไม่ระบุ  25.0                             อื่นๆ   
8558210     18/3/2022  ไม่ระบุ  25.0                             อื่นๆ   
8558211     18/3/2022  ไม่ระบุ  64.0                             อื่นๆ   
8558212     18/3/2022  ไม่ระบุ  64.0                             อื่นๆ   
8558213     18/3/2022  ไม่ระบุ  64.0                             อื่นๆ   

        province_of_onset  จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์)  \
0           กรุงเทพมหานคร            567335                   กลาง   
1           กรุงเทพมหานคร            567335                   กลาง   
2           กรุงเทพมหานคร            567335                   กลาง   
3           กรุงเทพมหานคร            567335                   กลาง   
4           กรุงเทพมหานคร            567335                   กลาง   
...                   ...               ...                    ...   
8558209     กรุงเทพมหานคร            567335                   กลาง   
8558210     กรุงเทพมหานคร            567335                   กลาง   
8558211         กำแพงเพชร             13042                   กลาง   
8558212         กำแพงเพชร             13042                   กลาง   
8558213         กำแพงเพชร             13042                   กลาง   

           ชื่อจังหวัด จำนวนประชากรชาย จำนวนประชากรหญิง  
0        กรุงเทพมหานคร         2592292          2935702  
1        กรุงเทพมหานคร         2592292          2935702  
2        กรุงเทพมหานคร         2592292          2935702  
3        กรุงเทพมหานคร         2592292          2935702  
4        กรุงเทพมหานคร         2592292          2935702  
...                ...             ...              ...  
8558209  กรุงเทพมหานคร         2592292          2935702  
8558210  กรุงเทพมหานคร         2592292          2935702  
8558211      กำแพงเพชร          351539           360604  
8558212      กำแพงเพชร          351539           360604  
8558213      กำแพงเพชร          351539           360604  

[8558214 rows x 10 columns]

### เปลี่ยน datetime เพื่อที่จะแบ่งข้อมูลออกไปใช้งาน

In [ ]:
import datetime
data_all_1 = pd.DataFrame(data_all)
data_all_1['announce_date'] = pd.to_datetime(data_all_1['announce_date'], format = "%d/%m/%Y")
data_all_1['year'] = data_all_1['announce_date'].dt.year
data_all_1['month'] = data_all_1['announce_date'].dt.month

In [ ]:
data_all_1

announce_date      sex   age                              risk  \
0          2020-01-12     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1          2020-01-12     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2          2020-01-12     หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3          2020-01-17     หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4          2020-01-17     หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
...               ...      ...   ...                               ...   
8558209    2022-03-18  ไม่ระบุ  25.0                             อื่นๆ   
8558210    2022-03-18  ไม่ระบุ  25.0                             อื่นๆ   
8558211    2022-03-18  ไม่ระบุ  64.0                             อื่นๆ   
8558212    2022-03-18  ไม่ระบุ  64.0                             อื่นๆ   
8558213    2022-03-18  ไม่ระบุ  64.0                             อื่นๆ   

        province_of_onset  จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์)  \
0           กรุงเทพมหานคร            567335                   กลาง   
1           กรุงเทพมหานคร            567335                   กลาง   
2           กรุงเทพมหานคร            567335                   กลาง   
3           กรุงเทพมหานคร            567335                   กลาง   
4           กรุงเทพมหานคร            567335                   กลาง   
...                   ...               ...                    ...   
8558209     กรุงเทพมหานคร            567335                   กลาง   
8558210     กรุงเทพมหานคร            567335                   กลาง   
8558211         กำแพงเพชร             13042                   กลาง   
8558212         กำแพงเพชร             13042                   กลาง   
8558213         กำแพงเพชร             13042                   กลาง   

           ชื่อจังหวัด จำนวนประชากรชาย จำนวนประชากรหญิง  year  month  
0        กรุงเทพมหานคร         2592292          2935702  2020      1  
1        กรุงเทพมหานคร         2592292          2935702  2020      1  
2        กรุงเทพมหานคร         2592292          2935702  2020      1  
3        กรุงเทพมหานคร         2592292          2935702  2020      1  
4        กรุงเทพมหานคร         2592292          2935702  2020      1  
...                ...             ...              ...   ...    ...  
8558209  กรุงเทพมหานคร         2592292          2935702  2022      3  
8558210  กรุงเทพมหานคร         2592292          2935702  2022      3  
8558211      กำแพงเพชร          351539           360604  2022      3  
8558212      กำแพงเพชร          351539           360604  2022      3  
8558213      กำแพงเพชร          351539           360604  2022      3  

[8558214 rows x 12 columns]

## ตารางรวมสุดท้าย

In [ ]:
data_final = data_all_1[['announce_date','year','month','sex','age','risk','province_of_onset','จำนวนผู้ติดเชื้อ','6 ภูมิภาค (ภูมิศาสตร์)','จำนวนประชากรชาย','จำนวนประชากรหญิง']]
data_final

announce_date  year  month      sex   age  \
0          2020-01-12  2020      1     หญิง  61.0   
1          2020-01-12  2020      1     หญิง  61.0   
2          2020-01-12  2020      1     หญิง  61.0   
3          2020-01-17  2020      1     หญิง  74.0   
4          2020-01-17  2020      1     หญิง  74.0   
...               ...   ...    ...      ...   ...   
8558209    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558210    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558211    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558212    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558213    2022-03-18  2022      3  ไม่ระบุ  64.0   

                                     risk province_of_onset  จำนวนผู้ติดเชื้อ  \
0        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
1        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
2        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
3        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
4        คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
...                                   ...               ...               ...   
8558209                             อื่นๆ     กรุงเทพมหานคร            567335   
8558210                             อื่นๆ     กรุงเทพมหานคร            567335   
8558211                             อื่นๆ         กำแพงเพชร             13042   
8558212                             อื่นๆ         กำแพงเพชร             13042   
8558213                             อื่นๆ         กำแพงเพชร             13042   

        6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย จำนวนประชากรหญิง  
0                         กลาง         2592292          2935702  
1                         กลาง         2592292          2935702  
2                         กลาง         2592292          2935702  
3                         กลาง         2592292          2935702  
4                         กลาง         2592292          2935702  
...                        ...             ...              ...  
8558209                   กลาง         2592292          2935702  
8558210                   กลาง         2592292          2935702  
8558211                   กลาง          351539           360604  
8558212                   กลาง          351539           360604  
8558213                   กลาง          351539           360604  

[8558214 rows x 11 columns]

## แบ่งข้อมูลไปใช้งาน

In [ ]:
data_1_20 = data_final[((data_final['month'] >= 1) & (data_final['month'] <= 12)) & (data_final['year'] == 2020)]
data_1_20

announce_date  year  month   sex   age  \
0        2020-01-12  2020      1  หญิง  61.0   
1        2020-01-12  2020      1  หญิง  61.0   
2        2020-01-12  2020      1  หญิง  61.0   
3        2020-01-17  2020      1  หญิง  74.0   
4        2020-01-17  2020      1  หญิง  74.0   
...             ...   ...    ...   ...   ...   
15536    2020-12-31  2020     12   ชาย  56.0   
15537    2020-12-31  2020     12   ชาย  56.0   
15538    2020-12-31  2020     12   ชาย  70.0   
15539    2020-12-31  2020     12   ชาย  70.0   
15540    2020-12-31  2020     12   ชาย  70.0   

                                   risk province_of_onset  จำนวนผู้ติดเชื้อ  \
0      คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
1      คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
2      คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
3      คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
4      คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร            567335   
...                                 ...               ...               ...   
15536                     Cluster ระยอง             ระยอง             75164   
15537                     Cluster ระยอง             ระยอง             75164   
15538                     Cluster ระยอง             ระยอง             75164   
15539                     Cluster ระยอง             ระยอง             75164   
15540                     Cluster ระยอง             ระยอง             75164   

      6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย จำนวนประชากรหญิง  
0                       กลาง         2592292          2935702  
1                       กลาง         2592292          2935702  
2                       กลาง         2592292          2935702  
3                       กลาง         2592292          2935702  
4                       กลาง         2592292          2935702  
...                      ...             ...              ...  
15536               ตะวันออก          368992           382351  
15537               ตะวันออก          368992           382351  
15538               ตะวันออก          368992           382351  
15539               ตะวันออก          368992           382351  
15540               ตะวันออก          368992           382351  

[15538 rows x 11 columns]

In [ ]:
data_0_21 = data_final[((data_final['month'] == 10)) & (data_all_1['year'] == 2021)]
data_0_21

announce_date  year  month   sex   age              risk  \
4118795    2021-10-01  2021     10  หญิง  30.0             อื่นๆ   
4118796    2021-10-01  2021     10  หญิง  30.0             อื่นๆ   
4118797    2021-10-01  2021     10  หญิง  30.0             อื่นๆ   
4118798    2021-10-01  2021     10  หญิง  32.0             อื่นๆ   
4118799    2021-10-01  2021     10  หญิง  32.0             อื่นๆ   
...               ...   ...    ...   ...   ...               ...   
4907151    2021-10-31  2021     10   ชาย  33.0  ทัณฑสถาน/เรือนจำ   
4907152    2021-10-31  2021     10   ชาย  60.0  ทัณฑสถาน/เรือนจำ   
4907153    2021-10-31  2021     10   ชาย  24.0  ทัณฑสถาน/เรือนจำ   
4907154    2021-10-31  2021     10  หญิง  31.0  ทัณฑสถาน/เรือนจำ   
4907155    2021-10-31  2021     10  หญิง  24.0  ทัณฑสถาน/เรือนจำ   

        province_of_onset  จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์)  \
4118795             พะเยา              5920                  เหนือ   
4118796             พะเยา              5920                  เหนือ   
4118797             พะเยา              5920                  เหนือ   
4118798             พะเยา              5920                  เหนือ   
4118799             พะเยา              5920                  เหนือ   
...                   ...               ...                    ...   
4907151           ไม่ระบุ            398327                ไม่ระบุ   
4907152           ไม่ระบุ            398327                ไม่ระบุ   
4907153           ไม่ระบุ            398327                ไม่ระบุ   
4907154           ไม่ระบุ            398327                ไม่ระบุ   
4907155           ไม่ระบุ            398327                ไม่ระบุ   

        จำนวนประชากรชาย จำนวนประชากรหญิง  
4118795          225943           238562  
4118796          225943           238562  
4118797          225943           238562  
4118798          225943           238562  
4118799          225943           238562  
...                 ...              ...  
4907151             NaN              NaN  
4907152             NaN              NaN  
4907153             NaN              NaN  
4907154             NaN              NaN  
4907155             NaN              NaN  

[788361 rows x 11 columns]

In [ ]:
#data_0_21.to_csv(os.path.join(path,'data_0_21.csv'))

In [ ]:
data_1_21 = data_final[((data_final['month'] == 11)) & (data_all_1['year'] == 2021)]
data_1_21.head()

announce_date  year  month   sex   age  \
4907156    2021-11-01  2021     11  หญิง  25.0   
4907157    2021-11-01  2021     11  หญิง  25.0   
4907158    2021-11-01  2021     11  หญิง  25.0   
4907159    2021-11-01  2021     11   ชาย  28.0   
4907160    2021-11-01  2021     11   ชาย  28.0   

                                                risk province_of_onset  \
4907156                                 สถานที่ทำงาน        แม่ฮ่องสอน   
4907157                                 สถานที่ทำงาน        แม่ฮ่องสอน   
4907158                                 สถานที่ทำงาน        แม่ฮ่องสอน   
4907159  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้        แม่ฮ่องสอน   
4907160  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้        แม่ฮ่องสอน   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
4907156              4145                  เหนือ          144811   
4907157              4145                  เหนือ          144811   
4907158              4145                  เหนือ          144811   
4907159              4145                  เหนือ          144811   
4907160              4145                  เหนือ          144811   

        จำนวนประชากรหญิง  
4907156           141105  
4907157           141105  
4907158           141105  
4907159           141105  
4907160           141105

In [ ]:
data_2_21 = data_final[((data_final['month'] == 12)) & (data_all_1['year'] == 2021)]
data_2_21.head()

announce_date  year  month   sex   age  \
5424602    2021-12-01  2021     12  หญิง  63.0   
5424603    2021-12-01  2021     12  หญิง  63.0   
5424604    2021-12-01  2021     12  หญิง  63.0   
5424605    2021-12-01  2021     12  หญิง  40.0   
5424606    2021-12-01  2021     12  หญิง  40.0   

                                                risk province_of_onset  \
5424602  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          เชียงราย   
5424603  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          เชียงราย   
5424604  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          เชียงราย   
5424605                                        อื่นๆ          เชียงราย   
5424606                                        อื่นๆ          เชียงราย   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
5424602              8122                  เหนือ          630868   
5424603              8122                  เหนือ          630868   
5424604              8122                  เหนือ          630868   
5424605              8122                  เหนือ          630868   
5424606              8122                  เหนือ          630868   

        จำนวนประชากรหญิง  
5424602           667557  
5424603           667557  
5424604           667557  
5424605           667557  
5424606           667557

In [ ]:
data_all_22 = data_final[((data_final['month'] == 1) | (data_final['month'] == 2) | (data_final['month'] == 3)) & (data_final['year'] == 2022)]
data_all_22

announce_date  year  month      sex   age  \
5701341    2022-01-01  2022      1      ชาย  39.0   
5701342    2022-01-01  2022      1      ชาย  39.0   
5701343    2022-01-01  2022      1      ชาย  39.0   
5701344    2022-01-01  2022      1     หญิง  39.0   
5701345    2022-01-01  2022      1     หญิง  39.0   
...               ...   ...    ...      ...   ...   
8558209    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558210    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558211    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558212    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558213    2022-03-18  2022      3  ไม่ระบุ  64.0   

                                                risk province_of_onset  \
5701341                                        อื่นๆ             พะเยา   
5701342                                        อื่นๆ             พะเยา   
5701343                                        อื่นๆ             พะเยา   
5701344  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
5701345  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
...                                              ...               ...   
8558209                                        อื่นๆ     กรุงเทพมหานคร   
8558210                                        อื่นๆ     กรุงเทพมหานคร   
8558211                                        อื่นๆ         กำแพงเพชร   
8558212                                        อื่นๆ         กำแพงเพชร   
8558213                                        อื่นๆ         กำแพงเพชร   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
5701341              5920                  เหนือ          225943   
5701342              5920                  เหนือ          225943   
5701343              5920                  เหนือ          225943   
5701344              5920                  เหนือ          225943   
5701345              5920                  เหนือ          225943   
...                   ...                    ...             ...   
8558209            567335                   กลาง         2592292   
8558210            567335                   กลาง         2592292   
8558211             13042                   กลาง          351539   
8558212             13042                   กลาง          351539   
8558213             13042                   กลาง          351539   

        จำนวนประชากรหญิง  
5701341           238562  
5701342           238562  
5701343           238562  
5701344           238562  
5701345           238562  
...                  ...  
8558209          2935702  
8558210          2935702  
8558211           360604  
8558212           360604  
8558213           360604  

[2856873 rows x 11 columns]

In [ ]:
data_1_22 = data_all_22[((data_all_22['month'] == 1) & (data_all_22['month'] == 1))]
data_1_22

announce_date  year  month   sex   age  \
5701341    2022-01-01  2022      1   ชาย  39.0   
5701342    2022-01-01  2022      1   ชาย  39.0   
5701343    2022-01-01  2022      1   ชาย  39.0   
5701344    2022-01-01  2022      1  หญิง  39.0   
5701345    2022-01-01  2022      1  หญิง  39.0   
...               ...   ...    ...   ...   ...   
6270517    2022-01-31  2022      1   ชาย  40.0   
6270518    2022-01-31  2022      1   ชาย  40.0   
6270519    2022-01-31  2022      1   ชาย  48.0   
6270520    2022-01-31  2022      1   ชาย  48.0   
6270521    2022-01-31  2022      1   ชาย  48.0   

                                                  risk province_of_onset  \
5701341                                          อื่นๆ             พะเยา   
5701342                                          อื่นๆ             พะเยา   
5701343                                          อื่นๆ             พะเยา   
5701344    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
5701345    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
...                                                ...               ...   
6270517  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว           นนทบุรี   
6270518  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว           นนทบุรี   
6270519    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          ปทุมธานี   
6270520    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          ปทุมธานี   
6270521    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          ปทุมธานี   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
5701341              5920                  เหนือ          225943   
5701342              5920                  เหนือ          225943   
5701343              5920                  เหนือ          225943   
5701344              5920                  เหนือ          225943   
5701345              5920                  เหนือ          225943   
...                   ...                    ...             ...   
6270517            117127                   กลาง          599167   
6270518            117127                   กลาง          599167   
6270519             85591                   กลาง          563851   
6270520             85591                   กลาง          563851   
6270521             85591                   กลาง          563851   

        จำนวนประชากรหญิง  
5701341           238562  
5701342           238562  
5701343           238562  
5701344           238562  
5701345           238562  
...                  ...  
6270517           689470  
6270518           689470  
6270519           626209  
6270520           626209  
6270521           626209  

[569181 rows x 11 columns]

In [ ]:
data_2_22=  data_all_22[((data_all_22['month'] == 2) & (data_all_22['month'] == 2))]
data_2_22

announce_date  year  month   sex   age  \
6270522    2022-02-01  2022      2  หญิง  35.0   
6270523    2022-02-01  2022      2  หญิง  35.0   
6270524    2022-02-01  2022      2  หญิง  35.0   
6270525    2022-02-01  2022      2  หญิง  19.0   
6270526    2022-02-01  2022      2  หญิง  19.0   
...               ...   ...    ...   ...   ...   
7462405    2022-02-28  2022      2  หญิง  37.0   
7462406    2022-02-28  2022      2  หญิง  37.0   
7462407    2022-02-28  2022      2  หญิง  40.0   
7462408    2022-02-28  2022      2  หญิง  40.0   
7462409    2022-02-28  2022      2  หญิง  40.0   

                                                risk province_of_onset  \
6270522                                        อื่นๆ             พะเยา   
6270523                                        อื่นๆ             พะเยา   
6270524                                        อื่นๆ             พะเยา   
6270525  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
6270526  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
...                                              ...               ...   
7462405                                        อื่นๆ     กรุงเทพมหานคร   
7462406                                        อื่นๆ     กรุงเทพมหานคร   
7462407                                        อื่นๆ     กรุงเทพมหานคร   
7462408                                        อื่นๆ     กรุงเทพมหานคร   
7462409                                        อื่นๆ     กรุงเทพมหานคร   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
6270522              5920                  เหนือ          225943   
6270523              5920                  เหนือ          225943   
6270524              5920                  เหนือ          225943   
6270525              5920                  เหนือ          225943   
6270526              5920                  เหนือ          225943   
...                   ...                    ...             ...   
7462405            567335                   กลาง         2592292   
7462406            567335                   กลาง         2592292   
7462407            567335                   กลาง         2592292   
7462408            567335                   กลาง         2592292   
7462409            567335                   กลาง         2592292   

        จำนวนประชากรหญิง  
6270522           238562  
6270523           238562  
6270524           238562  
6270525           238562  
6270526           238562  
...                  ...  
7462405          2935702  
7462406          2935702  
7462407          2935702  
7462408          2935702  
7462409          2935702  

[1191888 rows x 11 columns]

In [ ]:
data_3_22 = data_all_22[((data_all_22['month'] == 3) & (data_all_22['month'] == 3))]
data_3_22

announce_date  year  month      sex   age   risk province_of_onset  \
7462410    2022-03-01  2022      3     หญิง  47.0  อื่นๆ           สุโขทัย   
7462411    2022-03-01  2022      3     หญิง  47.0  อื่นๆ           สุโขทัย   
7462412    2022-03-01  2022      3     หญิง  47.0  อื่นๆ           สุโขทัย   
7462413    2022-03-01  2022      3     หญิง  34.0  อื่นๆ           สุโขทัย   
7462414    2022-03-01  2022      3     หญิง  34.0  อื่นๆ           สุโขทัย   
...               ...   ...    ...      ...   ...    ...               ...   
8558209    2022-03-18  2022      3  ไม่ระบุ  25.0  อื่นๆ     กรุงเทพมหานคร   
8558210    2022-03-18  2022      3  ไม่ระบุ  25.0  อื่นๆ     กรุงเทพมหานคร   
8558211    2022-03-18  2022      3  ไม่ระบุ  64.0  อื่นๆ         กำแพงเพชร   
8558212    2022-03-18  2022      3  ไม่ระบุ  64.0  อื่นๆ         กำแพงเพชร   
8558213    2022-03-18  2022      3  ไม่ระบุ  64.0  อื่นๆ         กำแพงเพชร   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
7462410             11847                   กลาง          283429   
7462411             11847                   กลาง          283429   
7462412             11847                   กลาง          283429   
7462413             11847                   กลาง          283429   
7462414             11847                   กลาง          283429   
...                   ...                    ...             ...   
8558209            567335                   กลาง         2592292   
8558210            567335                   กลาง         2592292   
8558211             13042                   กลาง          351539   
8558212             13042                   กลาง          351539   
8558213             13042                   กลาง          351539   

        จำนวนประชากรหญิง  
7462410           301923  
7462411           301923  
7462412           301923  
7462413           301923  
7462414           301923  
...                  ...  
8558209          2935702  
8558210          2935702  
8558211           360604  
8558212           360604  
8558213           360604  

[1095804 rows x 11 columns]

In [ ]:
data_Mar_1 = data_3_22[(data_3_22['announce_date'] >= '2022-03-11') & (data_3_22['announce_date'] <= '2022-03-18')]     # เลือกข้อมูล 1 อาทิตย์ล่าสุด
data_Mar_1 

announce_date  year  month      sex   age  \
8043724    2022-03-11  2022      3     หญิง  29.0   
8043725    2022-03-11  2022      3     หญิง  29.0   
8043726    2022-03-11  2022      3     หญิง  29.0   
8043727    2022-03-11  2022      3     หญิง  18.0   
8043728    2022-03-11  2022      3     หญิง  18.0   
...               ...   ...    ...      ...   ...   
8558209    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558210    2022-03-18  2022      3  ไม่ระบุ  25.0   
8558211    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558212    2022-03-18  2022      3  ไม่ระบุ  64.0   
8558213    2022-03-18  2022      3  ไม่ระบุ  64.0   

                                                risk province_of_onset  \
8043724  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้         เชียงใหม่   
8043725  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้         เชียงใหม่   
8043726  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้         เชียงใหม่   
8043727                         อยู่ระหว่างการสอบสวน         เชียงใหม่   
8043728                         อยู่ระหว่างการสอบสวน         เชียงใหม่   
...                                              ...               ...   
8558209                                        อื่นๆ     กรุงเทพมหานคร   
8558210                                        อื่นๆ     กรุงเทพมหานคร   
8558211                                        อื่นๆ         กำแพงเพชร   
8558212                                        อื่นๆ         กำแพงเพชร   
8558213                                        อื่นๆ         กำแพงเพชร   

         จำนวนผู้ติดเชื้อ 6 ภูมิภาค (ภูมิศาสตร์) จำนวนประชากรชาย  \
8043724             42424                  เหนือ          864364   
8043725             42424                  เหนือ          864364   
8043726             42424                  เหนือ          864364   
8043727             42424                  เหนือ          864364   
8043728             42424                  เหนือ          864364   
...                   ...                    ...             ...   
8558209            567335                   กลาง         2592292   
8558210            567335                   กลาง         2592292   
8558211             13042                   กลาง          351539   
8558212             13042                   กลาง          351539   
8558213             13042                   กลาง          351539   

        จำนวนประชากรหญิง  
8043724           925021  
8043725           925021  
8043726           925021  
8043727           925021  
8043728           925021  
...                  ...  
8558209          2935702  
8558210          2935702  
8558211           360604  
8558212           360604  
8558213           360604  

[514490 rows x 11 columns]

## โหลดข้อมูลออกไปใช้งาน

In [ ]:
data_1_20.to_csv(os.path.join(path,'data_1_20.csv')) #ข้อมูลปี2020

In [ ]:
data_1_21.to_csv(os.path.join(path,'data_1_21.csv')) # ข้อมูลเดือนพฤศจิกายน 2021

In [ ]:
data_2_21.to_csv(os.path.join(path,'data_2_21.csv')) # ข้อมูลเดือนธันวาคม 2021

In [ ]:
data_1_22.to_csv(os.path.join(path,'data_1_22.csv')) # ข้อมูลเดือนมกราคม 2022 มีขนาดใหญ่เกินไป

In [ ]:
data_3_22.to_csv(os.path.join(path,'data_3_22.csv')) # วันที่1-18 เดือนมีนาข้อมูลเยอะเกินไป

In [ ]:
data_Mar_1.to_csv(os.path.join(path,'data_Mar_1.csv'))   # ข้อมูลวันที่ 11-18 มีนาคม 2022 เป็นเวลา 7 วัน 1 อาทิตย์